src/tools/wasm-fuzz-lattices.cpp

```diff
@@ -0,0 +1,389 @@
+    analysis::FiniteIntPowersetLattice lattice(func->getNumLocals());
+    analysis::LivenessTransferFunction transferFunction;
+
+    check<analysis::FiniteIntPowersetLattice,
+          analysis::LivenessTransferFunction>(
+      cfg, lattice, transferFunction, "LivenessTransferFunction");
+  }
+
+  // Checks properties of the ReachingDefinitionsTransferFunction on a randomly
+  // generated module function.
+  void checkReachingDefinitionsTransferFunction(analysis::CFG& cfg,
+                                                Function* func) {
+    LocalGraph::GetSetses getSetses;
+    LocalGraph::Locations locations;
+    analysis::ReachingDefinitionsTransferFunction transferFunction(
+      func, getSetses, locations);
+    check<analysis::FinitePowersetLattice<LocalSet*>,
+          analysis::ReachingDefinitionsTransferFunction>(
+      cfg,
+      transferFunction.lattice,
+      transferFunction,
+      "ReachingDefinitionsTransferFunction");
+  }
+
+  // Generates a module. The module is used as an input to fuzz transfer
+  // functions as well as randomly generated lattice element states. Lattice
+  // properties are also fuzzed from the randomly generated states.
+  void run(uint64_t seed) {
+    // TODO: Reset the global type state to avoid monotonically increasing
+    // memory use.
+    RandEngine getRand(seed);
+    std::cout << "Running with seed " << seed << "\n";
+
+    // 4kb of random bytes should be enough for anyone!
+    std::vector<char> bytes(4096);
+    for (size_t i = 0; i < bytes.size(); i += sizeof(uint64_t)) {
+      *(uint64_t*)(bytes.data() + i) = getRand();
+    }
+
+    std::vector<char> bytes2 = bytes;
+
+    Module testModule;
+    TranslateToFuzzReader reader(testModule, std::move(bytes));
+    reader.build();
+    rand = Random(std::move(bytes2));
+
+    for (auto& func : testModule.functions) {
+      Function* currFunc = func.get();
+      // Imported functions can't be fuzzed here as they don't have a body.
+      if (currFunc->imported()) {
+        continue;
+      }
+      analysis::CFG cfg = analysis::CFG::fromFunction(currFunc);
+
+      checkLivenessTransferFunction(cfg, currFunc);
+      checkReachingDefinitionsTransferFunction(cfg, currFunc);
+    }
+  }
+};
+
+} // namespace wasm
+
+int main(int argc, const char* argv[]) {
+  using namespace wasm;
+
+  const std::string WasmFuzzTypesOption = "wasm-fuzz-lattices options";
+
+  Options options("wasm-fuzz-lattices",
+                  "Fuzz lattices for reflexivity, transitivity, and "
+                  "anti-symmetry, and tranfer functions for monotonicity.");
+
+  std::optional<uint64_t> seed;
+  options.add("--seed",
+              "",
+              "Run a single workload generated by the given seed",
+              WasmFuzzTypesOption,
+              Options::Arguments::One,
+              [&](Options*, const std::string& arg) {
+                seed = uint64_t(std::stoull(arg));
+              });
+
+  bool verbose = false;
```

Comment:
It doesn't look like you use the verbose flag for anything yet, but perhaps you could. See https://github.com/WebAssembly/binaryen/blob/main/test/lit/fuzz-types.test for an example of how to test that a fuzzer works using the verbose flag.